app/models/vacation.rb

```diff
@@ -86,7 +88,20 @@ def not_cancelled?
 
   def validate_start_date_close_to_weekend
     if start_date.thursday? || start_date.friday? || start_date.on_weekend?
-      errors.add(:start_date, I18n.t("activerecord.errors.models.vacation.attributes.start_date.close_weekend"))
+      errors.add(:start_date, :close_weekend)
+    end
+  end
+
+  def validate_start_date_close_to_holiday
+    return unless ENV['VALIDATE_VACATION_BEFORE_HOLIDAY'] == 'true'
+    if (user.office_holidays & restrict_dates).any?
+      errors.add(:start_date, :close_holiday)
+    end
+  end
+
+  def restrict_dates
+    (start_date..(start_date + MINIMUM_DAYS_BEFORE_HOLIDAY.days)).to_a.map do |date|
```

Comment:
I'm not quite sure, but I think this was misunderstood. According to CLT,  _§ 3 It is forbidden to start vacations in the period of two days before a public holiday or paid weekly rest day_
So, I think we would to like to prevent schedule vacations 2 days before a holiday (and the holiday itself) 

my suggestion to fix it:
```ruby
((start_date - MINIMUM_DAYS_BEFORE_HOLIDAY.days)..start_date).to_a
```